### PR TITLE
chore: Improve logging for sidecar

### PR DESCRIPTION
**Pull request by gjreda**
- https://github.com/refstudio/refstudio/issues/101

Temporarily logging to `/tmp` for now, but ideally we'd log this data in the $APPDATA directory. I don't think the sidecar has a way of knowing the $APPDATA path unless we pass it in.

```
❯ tail -f refstudio-sidecar.log
2023-06-06 20:25:23,575 - sidecar.ingest - INFO - Starting ingestion for project: project-x
2023-06-06 20:25:23,575 - sidecar.ingest - INFO - Input directory: /Users/greg/Library/Application Support/com.tauri.dev/project-x/uploads
2023-06-06 20:25:23,575 - sidecar.ingest - INFO - Found 2 pdf files to process
2023-06-06 20:25:23,575 - sidecar.ingest - INFO - Calling Grobid server...
2023-06-06 20:25:52,351 - sidecar.ingest - INFO - Finished calling Grobid server
2023-06-06 20:25:52,352 - sidecar.ingest - INFO - Grobid successfully parsed file: test.pdf
2023-06-06 20:25:52,352 - sidecar.ingest - WARNING - Grobid failed to parse file: FActScore-.Fine-grained.Atomic.Evaluation.of.Factual.Precision.in.Long.Form.Text.Generation.pdf
2023-06-06 20:25:52,352 - sidecar.ingest - INFO - Found 1 xml files to parse
2023-06-06 20:25:52,368 - sidecar.ingest - INFO - Found 1 json reference files
2023-06-06 20:25:52,368 - sidecar.ingest - INFO - Creating Reference from file: test.json
2023-06-06 20:25:52,369 - sidecar.ingest - INFO - Finished ingestion for project: project-x
2023-06-06 20:25:52,369 - sidecar.ingest - INFO - Response: {'project_name': 'project-x', 'references': [{'source_filename': 'test.pdf', 'filename_md5': '754dc77d28e62763c4916970d595a10f', 'title': 'A Few Useful Things to Know about Machine Learning', 'authors': [{'full_name': 'Pedro Domingos', 'given_name': 'Pedro', 'surname': 'Domingos', 'email': 'pedrod@cs.washington.edu'}]}]}
```

### CODE REVIEW

**Comment by cguedes** [comment]
> I don't think the sidecar has a way of knowing the $APPDATA path unless we pass it in.

@gjreda What do you think about creating a task that setup the sidecar with settings? This will be useful for sending paths ($APPDATA, ...) and also configurations (e.g. api keys).

File: python/sidecar/ingest.py [hunk 1]
```diff
@@ -4,21 +4,29 @@
...
 from .shared import HiddenPrints, chunk_text, get_filename_md5
 from .typing import Author, Reference
 
+logging.root.setLevel(logging.NOTSET)
+
 logger = logging.getLogger(__name__)
-logger.setLevel(logging.INFO)
+handler = logging.FileHandler(
+    os.path.join(
+        os.environ.get("SIDECAR_LOG_DIR", "/tmp"), "refstudio-sidecar.log"
```
Comment (cguedes) [inline python/sidecar/ingest.py:20]: shouldn't we call it `refstudio-sidecar-ingest.log` instead?

**Review by cguedes — CHANGES_REQUESTED** [review]

File: python/sidecar/ingest.py (hunk 1 above)
Comment (gjreda) [reply python/sidecar/ingest.py:20]: My intention is to have one unified log for the sidecar that includes both ingest and interactions.

**Review by cguedes — APPROVED** [review]

**Comment by gjreda** [comment]
> @gjreda What do you think about creating a task that setup the sidecar with settings? This will be useful for sending paths ($APPDATA, ...) and also configurations (e.g. api keys).

Yea, I agree. I think we're starting to see a need for some sort of broader settings configuration that can be shared by front and back end.

**Pull request merged**